zrml/court/src/lib.rs

```diff
@@ -20,44 +18,59 @@ mod pallet {
     use frame_support::{
         dispatch::DispatchResult,
         pallet_prelude::StorageMap,
-        traits::{Currency, Hooks, IsType},
+        traits::{Currency, Get, Hooks, IsType, Randomness, ReservableCurrency},
         Blake2_128Concat,
     };
     use frame_system::{ensure_signed, pallet_prelude::OriginFor};
-    use sp_runtime::DispatchError;
+    use sp_runtime::{traits::Saturating, ArithmeticError, DispatchError, SaturatedConversion};
+    use zeitgeist_primitives::{
+        traits::DisputeApi,
+        types::{Market, OutcomeReport, ResolutionCounters},
+    };
     use zrml_market_commons::MarketCommonsPalletApi;
 
+    const MAX_RANDOM_JURORS: usize = 5;
+
     pub(crate) type BalanceOf<T> =
         <CurrencyOf<T> as Currency<<T as frame_system::Config>::AccountId>>::Balance;
     pub(crate) type CurrencyOf<T> =
         <<T as Config>::MarketCommons as MarketCommonsPalletApi>::Currency;
+    pub(crate) type MarketIdOf<T> =
+        <<T as Config>::MarketCommons as MarketCommonsPalletApi>::MarketId;
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
+        #[frame_support::transactional]
```

Comment:
Does this need to be marked `transactional`?

Reply:
Nope :)